src/fsharp/DotNetFrameworkDependencies.fs

```diff
@@ -34,114 +34,78 @@ module internal FSharp.Compiler.DotNetFrameworkDependencies
+            let microsoftNETCoreAppRef = Path.Combine(implementationAssemblyDir, "../../../packs/Microsoft.NETCore.App.Ref")
+            if Directory.Exists(microsoftNETCoreAppRef) then
+                Some version, Some microsoftNETCoreAppRef
             else
-                None
-        | _ -> None
+               Some version,  None
+        with | _ -> None, None
 
     let isInReferenceAssemblyPackDirectory filename =
-        match getFrameworkRefsPackDirectoryPath with
-        | Some appRefDir ->
+        match frameworkRefsPackDirectoryRoot with
+        | Some root ->
             let path = Path.GetDirectoryName(filename)
-            path.StartsWith(appRefDir, StringComparison.OrdinalIgnoreCase)
+            path.StartsWith(root, StringComparison.OrdinalIgnoreCase)
         | _ -> false
 
-    let getFrameworkRefsPackDirectory =
-        match executionTfm, getFrameworkRefsPackDirectoryPath with
-        | Some tfm, Some appRefDir ->
+    let frameworkRefsPackDirectory =
+        let tfmPrefix = "netcoreapp"
+        let tfmCompare c1 c2 =
+            let deconstructTfmApp (netcoreApp: DirectoryInfo) =
+                let name = netcoreApp.Name
+                try
+                    if name.StartsWith(tfmPrefix, StringComparison.InvariantCultureIgnoreCase) then
+                        Some (Double.Parse(name.Substring(tfmPrefix.Length)))
+                    else
+                        None
+                with _ -> None
+
+            if c1 = c2 then 0
+            else
+                match (deconstructTfmApp c1), (deconstructTfmApp c2) with
+                | Some c1, Some c2 -> int(c1 - c2)
+                | None, Some _ -> -1
+                | Some _, None -> 1
+                | _ -> 0
+
+        match version, frameworkRefsPackDirectoryRoot with
+        | Some version, Some root ->
             try
-                let refDirs = Directory.GetDirectories(appRefDir)
-                let versionPath = refDirs |> Array.sortWith (versionCompare) |> Array.last
-                Some(Path.Combine(versionPath, "ref", tfm))
+                let ref = Path.Combine(root, version, "ref")
+                let highestTfm = DirectoryInfo(ref).GetDirectories()
+                                 |> Array.sortWith tfmCompare
+                                 |> Array.tryLast
+
+                match highestTfm with
+                | Some tfm -> Some (Path.Combine(ref, tfm.Name))
+                | None -> None
             with | _ -> None
         | _ -> None
 
-
-
     let getDependenciesOf assemblyReferences =
         let assemblies = new Dictionary<string, string>()
 
         // Identify path to a dll in the framework directory from a simple name
         let frameworkPathFromSimpleName simpleName =
-            let pathDll = Path.Combine(frameworkDir, simpleName + ".dll")
+            let pathDll = Path.Combine(implementationAssemblyDir, simpleName + ".dll")
```

Comment:
Will the extension always be .dll or .exe?  What about Linux?

Reply:
@brettfo Linux has no extension, so would be found by line 145:

This code covers the case when the assembly name does not match the file name because of the file extension.
````
           if not (File.Exists(pathDll)) then
                let pathExe = Path.Combine(frameworkDir, simpleName + ".exe")
                let pathExe = Path.Combine(implementationAssemblyDir, simpleName + ".exe")
                if not (File.Exists(pathExe)) then
                    pathDll
                else
                    pathExe
            else
                pathDll

````